src/bentoml/_internal/service/service.py

```diff
@@ -226,6 +230,40 @@ def __init__(
             models=[] if models is None else models,
         )
 
+        # Set import origin info - import_str can not be determined at this stage yet as
+        # the variable name is only available in module vars after __init__ is returned
+        # get_service_import_origin below will use the _caller_module for retriving the
+        # correct import_str for this service
+        caller_module = inspect.currentframe().f_back.f_globals["__name__"]
+        object.__setattr__(self, "_caller_module", caller_module)
+        object.__setattr__(self, "_working_dir", os.getcwd())
+
+    def get_service_import_origin(self) -> tuple[str, str]:
+        """
+        Returns the module name and working directory of the service
+        """
+        if not self._import_str:
+            import_module = self._caller_module
+            if import_module == "__main__":
+                if hasattr(sys.modules["__main__"], "__file__"):
+                    import_module = sys.modules["__main__"].__file__
+
+            for name, value in vars(sys.modules[self._caller_module]).items():
```

Comment:
This is a bit hacky, ideally, we should get the pickle serialization to work properly for bentoml.Service objects.

Reply:
This seems fine actually. I don't think it is that hacky.

Reply:
Added a few error messages to show the user the limit of this approach. Basically we require the user to put the Service object at module's top level for it to work. E.g. using a function that returns a Service instance may not work properly here.